src/components/select/select.template.tsx

```diff
@@ -246,21 +246,42 @@ function isSelected(option: Option, value: Option | Option[]): boolean {
 
 function createMenuItems(
     options: Option[],
-    value: Option | Option[]
+    value: Option | Option[],
+    selectIsRequired = false,
 ): Array<ListItem<Option>> {
-    return options.map((option) => {
-        const selected = isSelected(option, value);
-        const { text, disabled } = option;
-
-        return {
-            text: text,
-            selected: selected,
-            disabled: disabled,
-            value: option,
-            icon: option.icon,
-            iconColor: option.iconColor,
-        };
-    });
+    const menuOptionFilter = getMenuOptionFilter(selectIsRequired);
+    return options
+        .filter(menuOptionFilter)
+        .map((option) => {
+            const selected = isSelected(option, value);
+            const { text, disabled } = option;
+
+            return {
+                text: text,
+                selected: selected,
+                disabled: disabled,
+                value: option,
+                icon: option.icon,
+                iconColor: option.iconColor,
+            };
+        });
+}
+
+function getMenuOptionFilter(selectIsRequired) {
```

Comment:
Just to make it more readable.

```suggestion
function getMenuOptionFilter(selectIsRequired: boolean) {
```

Reply:
⚡